lib/public/AppFramework/Db/SnowflakeAwareEntity.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace OCP\AppFramework\Db;
+
+use OC\Snowflake\Decoder;
+use OC\Snowflake\Generator;
+use OCP\Server;
+
+class SnowflakeAwareEntity extends Entity {
+
```

Comment:
```suggestion

    private int $id;
```

```suggestion

    /** @var int $id */
    private $id;
```

One of this should silence psalm warnings

Reply:
It's not an `int` tho, at least on the entity level? https://github.com/nextcloud/documentation/pull/13811